node/src/syncer.rs

```diff
@@ -79,22 +84,25 @@ where
     S: Store,
 {
     pub fn start(args: SyncerArgs<S>) -> Result<Self> {
+        let cancellation_token = CancellationToken::new();
         let (cmd_tx, cmd_rx) = mpsc::channel(16);
-        let mut worker = Worker::new(args, cmd_rx)?;
+        let mut worker = Worker::new(args, cancellation_token.child_token(), cmd_rx)?;
 
         spawn(async move {
             worker.run().await;
         });
 
         Ok(Syncer {
+            cancellation_token,
             cmd_tx,
             _store: PhantomData,
         })
     }
 
-    pub async fn stop(&self) -> Result<()> {
-        // TODO
-        Ok(())
+    pub fn stop(&self) {
+        // Singal the Worker to stop.
+        // TODO: Should we wait for the Worker to stop?
```

Comment:
I think that's a good idea in general, not necesarily in this PR, if it's a hassle.

Reply:
it is a bit, that is why i have it as todo